girder/models/model_base.py

```diff
@@ -791,6 +791,15 @@ def _setAccess(self, doc, id, entity, level, save, flags=None, user=None, force=
             if '_id' not in doc:
                 doc = self.save(doc)
             else:
+                updateType = update.keys()[0]
+                # copy all other (potentially updated) fields to the update list
+                if updateType != '$set':
```

Comment:
For readability, let's invert this conditional since double negation can be confusing.